### PR TITLE
Remove `_requester` from MedicationRequest

### PR DESCRIPTION
**Pull request by KevinCranmer**
The Extension that was in _requester should now go in requester.extension

### CODE REVIEW

**Review by bcastillomartel-va — DISMISSED** [review]

File: fhir-resource-conversion/src/test/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/MedicationOrderConverterTest.java [hunk 1]
```diff
@@ -44,11 +46,10 @@ void dataAbsentReason() {
...
         MedicationOrderSamples.r4()
             .medicationRequest()
-            .requester(null)
-            ._requester(DataAbsentReason.of(DataAbsentReason.Reason.unknown));
-    assertThat(json(MedicationOrderConverter.INSTANCE.convert(r4Sample))).isEqualTo(json(dstu2));
-    // Create new DAR
-    r4Sample._requester(null);
+            .requester(
+                Reference.builder()
+                    .extension(List.of(DataAbsentReason.of(DataAbsentReason.Reason.unknown)))
```
Comment (KevinCranmer) [inline fhir-resource-conversion/src/test/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/MedicationOrderConverterTest.java:51]: This ends up looking like `extension[0].extension[0].DAR`. Is that what we want? Seems goofy

File: us-core-r4-examples/src/main/java/gov/va/api/health/r4/api/swaggerexamples/SwaggerMedicationRequest.java [hunk 2]
```diff
@@ -37,12 +37,16 @@ public static MedicationRequest medicationRequest() {
...
-        ._requester(
-            Extension.builder()
+        .requester(
+            Reference.builder()
                 .extension(
                     Extension.builder()
-                        .url("https://hl7.org/fhir/extension-data-absent-reason.html")
-                        .valueCode("unknown")
+                        .extension(
+                            Extension.builder()
```
Comment (jblefkove-va) [inline us-core-r4-examples/src/main/java/gov/va/api/health/r4/api/swaggerexamples/SwaggerMedicationRequest.java:45]: @jhulbert-va do you know why we did extension-within-extension previously? I would expect this to be:

```
.requester(
  Reference.builder()
    .extension(
      Extension.builder()
        .url("http://hl7.org/fhir/StructureDefinition/data-absent-reason")
        .valueCode("unknown")
        .build()
        .asList())
    .build())
```

Comment (jhulbert-va) [reply us-core-r4-examples/src/main/java/gov/va/api/health/r4/api/swaggerexamples/SwaggerMedicationRequest.java:45]: Yes, I think that's what it should look like. The fhir spec seems to define it as such also:
http://hl7.org/fhir/r4/extension-data-absent-reason.html

File: fhir-resource-conversion/src/main/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/MedicationOrderConverter.java [hunk 3]
```diff
@@ -37,10 +37,8 @@ public abstract class MedicationOrderConverter
   @Mapping(target = "dateWritten", source = "authoredOn")
   @Mapping(target = "note", qualifiedByName = "convertNote")
   @Mapping(target = "patient", source = "subject")
-  @Mapping(target = "prescriber", source = "requester")
-  @Mapping(
-      target = "_prescriber",
-      expression = "java(supportRequesterOrDar(source.requester(), source._requester()))")
+  @Mapping(target = "prescriber", source = "requester", qualifiedByName = "supportRequesterOrNull")
+  @Mapping(target = "_prescriber", source = "requester", qualifiedByName = "supportRequesterOrDar")
```
[the remark this answers is not in the document]
Comment (jhulbert-va) [reply fhir-resource-conversion/src/main/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/MedicationOrderConverter.java:41]: No, if the field doesn't exist, you can just ignore it.

[the remark this answers is not in the document]
Comment (jblefkove-va) [reply fhir-resource-conversion/src/main/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/MedicationOrderConverter.java:41]: Oh, I mixed up `_requester` and `_prescriber`. `_prescriber` is still supported in DSTU2 `MedicationOrder`

**Review by jhulbert-va — APPROVED** [review]

**Review by jblefkove-va — APPROVED** [review]

**Pull request merged**